data_handler/handlers/liquidable_debt/debt_handlers.py

```diff
@@ -49,45 +76,189 @@ def prepare_data(self, protocol_name: str, path: str = LOCAL_STORAGE_PATH) -> di
+        # state.collateral_interest_rate_models = InterestRateModels(
+        #     values=interest_rate_models[COLLATERAL_FIELD_NAME.lower()]
+        # )
+        # state.debt_interest_rate_models = InterestRateModels(
+        #     values=interest_rate_models[DEBT_FIELD_NAME.lower()]
+        # )
+        # for index, row in data.items():
+            # even if it isn't per-user data, we need to provide a user ID
+            # so like that we're able to provide debt and collateral values
+            # TODO use just one collateral token and debt token
+            # for collateral_token in row[COLLATERAL_FIELD_NAME].keys():
+            #     hypothetical_collateral_token_prices = self.get_prices_range(
+            #         collateral_token_name=collateral_token,
+            #         current_price=current_prices.prices.values[collateral_token]
+            #     )
 
-            state.loan_entities[data[row_number][USER_FIELD_NAME]].debt.values = {
-                key: value
-                for key, value in data[row_number][DEBT_FIELD_NAME].items()
-            }
-            state.loan_entities[data[row_number][USER_FIELD_NAME]].collateral.values = {
-                key: value
-                for key, value in data[row_number][COLLATERAL_FIELD_NAME].items()
-            }
+        # Go through first hypothetical prices and then through the debts
+        for hypothetical_price in hypothetical_collateral_token_prices:
+            current_prices.prices.values[collateral_token] = hypothetical_price
 
-            for token in data[row_number][DEBT_FIELD_NAME]:
-                if not data[row_number][COLLATERAL_FIELD_NAME].get(token, ""):
-                    continue
+            if not isinstance(state, (HashstackV0State, HashstackV1State)):
+                # FIXME not needed, will be done in the first step
+                # state.loan_entities[user_wallet_id].debt.values = {
+                #     debt_token: hypothetical_price
+                #     for debt_token in TOKEN_PAIRS.get(collateral_token, "")
+                # }
+                # state.loan_entities[user_wallet_id].collateral.values = {
+                #     collateral_token: hypothetical_price
+                # }
 
-                result = state.compute_liquidable_debt_at_price(
-                    prices=TokenValues(),
-                    collateral_token=token,
-                    collateral_token_price=data[row_number][COLLATERAL_FIELD_NAME][token],
-                    debt_token=token,
-                    risk_adjusted_collateral_usd=data[row_number][RISK_ADJUSTED_COLLATERAL_USD_FIELD_NAME],
-                    debt_usd=data[row_number][DEBT_USD_FIELD_NAME],
-                    health_factor=data[row_number][HEALTH_FACTOR_FIELD_NAME],
-                )
+            liquidable_debt = state.compute_liquidable_debt_at_price(
+                prices=TokenValues(init_value=current_prices.prices.values.get(collateral_token)),
```

Comment:
```suggestion
                prices=current_prices.prices,
```
`state.compute_liquidable_debt_at_price` needs the prices of all tokens.